### ISSUE TITLE
Practice Test 3 - Path coverage question

### ISSUE THREAD

**Issue by Anvita2110** [nus-cs2103-AY2425S1/forum#869]
### How would the answer change if it asked for statement coverage instead of path coverage in this question?

---

![image](https://github.com/user-attachments/assets/85552275-6097-42ea-ac6b-3fbe0c036e16)

**Comment by emmannyyy** [comment]
Not 100% sure, but I think that it should be 2 if the question was asking for statement coverage. 

[null] and [any string with length >0] can be used to reach all statements in the code.

**Comment by jagdeepsh** [comment]
Interesting question, I tried answering this myself while referencing the definition of statement coverage as attached below. If let's say, their definition of `execution of code` is something like how each line is running, something like the IntelliJ debugging step tool. I think the answer might be `3`. The smallest test case I can think of would be a string with length 0, `String s == ""` but not a string that is `null`.

Then I think line 1, 2, 6 will be executed. But the program does nothing afterwards, because of the for loop condition. If the string was null, then the lines of code executed would be 4, which would be line 1, 2, 3, 4, which would not be the smallest lines of code.

I am not sure if this is correct or if I myself misunderstood the concept/question. But this was what I came up with. Hope this helps :)

<img width="728" alt="Screenshot 2024-11-24 at 9 54 39 AM" src="https://github.com/user-attachments/assets/7a7b2707-7129-4ddf-89b0-bcfa5255f6e7">
